data/transaction/transaction.go

```diff
@@ -79,14 +85,21 @@ func (tx *Transaction) GetDataForSigning(encoder data.Encoder, marshaller data.M
 		return nil, err
 	}
 
-	if tx.InnerTransaction != nil {
-		ftx.InnerTransaction, err = tx.InnerTransaction.prepareTx(encoder)
-		if err != nil {
-			return nil, err
+	numInnerTxs := len(tx.InnerTransactions)
+	if numInnerTxs > 0 {
+		ftx.InnerTransactions = make([]*FrontendTransaction, numInnerTxs)
+		for i := 0; i < numInnerTxs; i++ {
+			innerFtx, errPrepare := tx.InnerTransactions[i].prepareTx(encoder)
```

Comment:
can you directly use ftx.InnerTransactions[i] on the assignment here so that we decrease the number of allocations.

Reply:
right, pushed